plugins/modules/mso_schema_site_anp_epg_domain.py

```diff
@@ -222,14 +263,23 @@ def main():
         switching_mode=dict(type="str"),
         enhanced_lagpolicy_name=dict(type="str"),
         enhanced_lagpolicy_dn=dict(type="str"),
+        binding_type=dict(type="str", default="none", choices=["dynamic", "ephemeral", "none", "static"]),
+        port_allocation=dict(type="str", choices=["elastic", "fixed"]),
+        num_ports=dict(type="int", default=0, min=0, max=65535),
+        netflow_pref=dict(type="str", default="disabled", choices=["enabled", "disabled"]),
+        allow_promiscuous=dict(type="str", default="reject", choices=["accept", "reject"]),
+        forged_transmits=dict(type="str", default="reject", choices=["accept", "reject"]),
+        mac_changes=dict(type="str", default="reject", choices=["accept", "reject"]),
+        delimiter=dict(type="str", choices=["+", "|", "~", "!", "@", "^", "="]),
+        custom_epg_name=dict(type="str"),
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_if=[
             ["state", "absent", ["domain_association_type", "domain_profile", "deployment_immediacy", "resolution_immediacy"]],
-            ["state", "present", ["domain_association_type", "domain_profile", "deployment_immediacy", "resolution_immediacy"]],
+            ["state", "present", ["domain_association_type", "domain_profile", "deployment_immediacy", "resolution_immediacy", "binding_type", "netflow_pref", "allow_promiscuous", "forged_transmits", "mac_changes"]],
```

Comment:
Making these new attributes required on present is version depending and will likely break existing functionality. So don't think these should be made required on present. If we want to catch this we would need to do a version check. What is the error response from NDO when it is not provided in 4.2.x?

Reply:
Yes I think we shouldn't put it in required to don't break other versions. 
The error message is the following: MSO Error 400: site-local EPG XX in site XX in template XX: BindingType is required for VMware VMM Domain and is the same for other attributes.